### ISSUE TITLE
Accessibility: Element is not keyboard-operable – <span> instead of buttons

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#51826]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The element(s) are not keyboard-focusable or clickable without a mouse because they are coded as <span> instead of HTML buttons.

![T002_ShowHideIconIsNotReachableThroughKB](https://github.com/user-attachments/assets/accfe1b6-8d00-420b-a727-31157996c843)

### Component or Page: 

Front-end Login Form on My Account Page

### Name affected element:

Show/Password icon

### Relevant code snippets:

`<span class="show-password-input"></span>`

### Recommended fix:

Instead of using the <span> element, it is recommended to use HTML <button> tags for these elements to make them inherently operable with keyboard.
 
 If you cannot replace a span with a button, it is possible to remediate it to function as a button with ARIA. The span would need to be given a button role and tabindex=""0"" to ensure that it can be focused with the keyboard. You would also need to write JavaScript to ensure that the span will function properly as a button for keyboard users. For more information on the aria button role, see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by darrenjacoby** [comment]
@dkotter good point, I've updated the PR to add `i18n_password_show` and `i18n_password_hide`. I'm not sure if this is a desirable structure (under namespace `woocommerce`), so let me know your thoughts.
